src/_guides/libraries/java-interop.md

```diff
@@ -0,0 +1,133 @@
+---
+title: "Java interop using package:jnigen"
+description: "To use Java in your Dart program, use package:jnigen."
+jnigen: "https://pub.dev/packages/jnigen"
+jni: "https://pub.dev/packages/jni"
+example: "https://github.com/HosseinYousefi/jnigen_example/tree/main"
+jnidoc: "https://docs.oracle.com/en/java/javase/17/docs/specs/jni/index.html"
+---
+
+Dart mobile, command-line, and server apps
+running on the [Dart Native platform](/overview#platform), on Android, Windows,
+macOS, and Linux can use [`package:jni`]({{page.jni}})
+and [`package:jnigen`]({{page.jnigen}})
+to call Java and Kotlin APIs.
+
+{{site.alert.note}}
+  This interop feature is **experimental**,
+  and [in active development](https://github.com/dart-lang/sdk/issues/49674).
+{{site.alert.end}}
+
+`package:jni` allows Dart code to interact
+with Java through [JNI]({{page.jnidoc}}). 
+However, doing so involves a lot of boilerplate code,
+so you can use `package:jnigen` to automatically generate
+the Dart bindings for a given Java API.
+
+You can decompile Kotlin to Java, allowing `package:jnigen`
+to generate bindings for Kotlin as well.
```

Comment:
It can be used on both Java byte code and Java source. Kotlin needs to be first compiled to Java byte code to be usable by jnigen.

I plan to add that example later.

Reply:
Sounds good, thanks!

Should this say the following then?

```suggestion
You can compile Kotlin to Java bytecode, allowing `package:jnigen`
to generate bindings for Kotlin as well.
```